### ISSUE TITLE
Block-Enabled Widget Area

### ISSUE THREAD

**Issue by brucedwilliams** [WordPress/gutenberg#25800]
**Describe the bug**
The new block-enabled widget area with Gutenberg 9.1 is a disaster. Very few of the previous widgets carry-over to the block-enabled widget implementation which causes some serious headaches for the end-user. Overall, the transition to the block-enabled widget area needs to handled much better before this feature of Gutenberg is enabled for the standard WordPress install.

**To reproduce**
1. Install Gutenberg 9.1

**Expected behavior**
The expectation is that prior widgets will be usable in the new block-enabled widget area via a shortcode or a "classic widget" block or some such. Throwing out all the non-block-enabled widgets is not a workable solution.

Additionally, why can't I use ALL blocks in the widget area? There seems to be an arbitrary sub-set of blocks that appear in the block-enabled widget area. What determines which blocks can be used in the block-enabled widget area and which can't???

**Screenshots**

**Editor version (please complete the following information):**
- WordPress version: 5.5.1
- Does the website has Gutenberg plugin installed, or is it using the block editor that comes by default? Gutenberg plugin
- If the Gutenberg plugin is installed, which version is it? 9.1

**Desktop (please complete the following information):**
 - OS: MacOS 10.15.6
 - Browser: Safari
 - Version: 14.0

**Comment by draganescu** [comment]
Howdy @brucedwilliams! Would you mind providing a set of steps to reproduce the problem you are describing? Which blocks are you missing in the widgets editor? Which widgets were lost? Do they work if you disable the Gutenberg plugin?

Thank you for the patience to test and for coming back with more thorough reports! 😄

**Comment by brucedwilliams** [comment]
I'm happy to give you more context on this issue. I will use a new site that I'm currently working on as an example. With WordPress 5.5.1 and WITHOUT the Gutenberg 9.1 plugin activated, the widgets available are displayed thusly:

![widgets_gutenberg_91_disabled](https://user-images.githubusercontent.com/72280473/95487146-ee55a500-0961-11eb-97f8-5e39526dbeb4.png)

and when Gutenberg 9.1 is enabled, the extremely restricted list of widgets available is as follows:

![widgets_gutenberg_91_enabled](https://user-images.githubusercontent.com/72280473/95487218-062d2900-0962-11eb-819b-f5781278ffa6.png)

Thank you for your patience in doing quality control and for coming back with more thoroughly tested plugins! 😉

**Comment by draganescu** [comment]
@brucedwilliams I did not mean to thank you in a sarcastic manner, I really meant it :). 

I will test with the bbpress widgets, but wanted to point out that the list in the inline inserter is made of the most recently used blocks. To see a full list of blocks available use the blue + (plus) icon in the top bar, next to the screen's title.

That should open the inserter as a sidebar and you can see the full list of available blocks and legacy widgets.

**Comment by brucedwilliams** [comment]
Ah. Your response highlights what is the likely issue here: the UI for inserting a block via the black "+" icon at the right-hand side of a new line in the block editor has changed it's behavior.

Using WordPress 5.5.1 without the Gutenberg plugin activated, clicking the black "+" icon at the right-hand side of a new line brings up a dialog like this:

![gutenberg_browse_all](https://user-images.githubusercontent.com/72280473/95586702-09cab980-0a0f-11eb-9ecf-24a25f270c4e.png)

Note the "Browse all" button at the bottom of the dialog.

With Gutenberg 9.1.x activated, this same dialog is missing the "Browse all" button (see more screen capture above) but apparently only when using it in the context of editing widgets. When one is using the Gutenberg editor with posts/pages, the "Browse all" button is still included. This makes the behavior of this UI element inconsistent when it is used during the "Widgets" editing process versus elsewhere. This is a "bad thing"...

BTW, I literally NEVER use the blue "+" icon in the top bar as it is pretty much guaranteed to be more mousing to get to it than to the black "+" icon at the end of a new line. Good UI design seeks to minimize the mouse input necessary to access features.

Finally, I actually really dislike the blue "+" icon in the top bar in the first place. It is readily apparent that you are "adding" _something_ by clicking it, but adding WHAT? At first, I literally assumed that the blue "+" icon in the top bar of the block editor added a new post/page similar to the Classic editor's "Add New" button (placement is similar after all). Please consider adding some kind of graphic/textual indication of WHAT is being added to the "+" icons, both black and blue.

I wouldn't mind an additional "+" icon in the Gutenberg editor that allowed a quick method of creating a new post/page (i.e., include the "Add New" functionality from the Classic editor). The only way to add a new post/page with Gutenberg is to go to the Dashboard menu which involves YET MORE mousing around the interface and opening of sub-menus.

Thanks!

**Comment by paaljoachim** [comment]
Hi Bruce @brucedwilliams 

Thank you for the feedback! It is very helpful! As it gets me to think of how we can perhaps adjust existing features to make the even more user friendly.

In relation to the inline inserter. I made a new issue so that we have a consistent treatment of the Widget screen inline inserter in relation with the Gutenberg inline inserter.

In relation to clicking the + icon in the top. It is interesting as you say that it is in the same location as adding a new post/page in the Classic editor. It would be great with even more detailed feedback. 
For instance what would be your ideal way to add a block to the layout?

> Good UI design seeks to minimize the mouse input necessary to access features.

I agree.

**Comment by brucedwilliams** [comment]
I would hope that the "inline inserter" is implemented only once and then called (with appropriate arguments) from both the standard Gutenberg editor and the new Gutenberg widget editor...

When I referenced my assumption that the "+" icon in the top bar of the Gutenberg editor created a new document not a new block and compared to the "Add New" button in the classic editor, I was referring to this UI element:

![classic_editor_add_new](https://user-images.githubusercontent.com/72280473/95756043-3f6fdc80-0c73-11eb-81a8-5c24061af3f5.jpg)

While one could argue that the "Add New" text is also ambiguous (add new WHAT?), it is an established WordPress editor interface element that experienced users are familiar with.

For the Gutenberg editor, I would use UI elements that incorporate both an icon along with text to give users who like graphic IU elements and also users who like textual elements something to like. A starting point for appropriate icons might be something like the Font Awesome "[file-plus](https://fontawesome.com/icons/file-plus?style=light)" icon for a new page/post/etc and something visually distinctive for a new block (I have no good ideas for you there). Both of these would live in the top bar.

Someone more versed in actual UI design could likely mock up a better concept than what I've described above...

**Comment by paaljoachim** [comment]
There are some ideas for Full Site Editing of using the left sidebar for adding new pages/posts. We will see along the way how things happen.

Thank you Bruce @brucedwilliams for sharing your thoughts!
It is greatly appreciated!
